priam/src/main/java/com/netflix/priam/resources/BackupServlet.java

```diff
@@ -135,43 +131,24 @@ public Response status() throws Exception {
     @Produces(MediaType.APPLICATION_JSON)
     public Response statusByDate(@PathParam("date") String date) throws Exception {
         JSONObject object = new JSONObject();
-        List<BackupMetadata> metadataLinkedList = this.completedBkups.locate(date);
-
-        if (metadataLinkedList != null && !metadataLinkedList.isEmpty()) {
-            // backup exist base on requested date, lets fetch more of its metadata
-            BackupMetadata bkupMetadata = metadataLinkedList.get(0);
-            object.put("Snapshotstatus", bkupMetadata.getStatus().equals(Status.FINISHED));
-            String token = bkupMetadata.getToken();
-            if (token != null && !token.isEmpty()) {
-                object.put("token", bkupMetadata.getToken());
-            } else {
-                object.put("token", "not available");
-            }
-            if (bkupMetadata.getStart() != null) {
-                object.put("starttime", DateUtil.formatyyyyMMddHHmm(bkupMetadata.getStart()));
-            } else {
-                object.put("starttime", "not available");
-            }
-
-            if (bkupMetadata.getCompleted() != null) {
-                object.put(
-                        "completetime", DateUtil.formatyyyyMMddHHmm(bkupMetadata.getCompleted()));
-            } else {
-                object.put("completetime", "not_available");
-            }
-
-        } else { // Backup do not exist for that date.
+        Optional<BackupMetadata> backupMetadataOptional =
+                this.completedBkups
+                        .locate(date)
+                        .stream()
+                        .filter(backupMetadata -> backupMetadata != null)
+                        .filter(backupMetadata -> backupMetadata.getStatus() == Status.FINISHED)
+                        .filter(
+                                backupMetadata ->
+                                        backupMetadata.getBackupVersion()
+                                                == SnapshotBackup.BACKUP_VERSION)
+                        .findFirst();
+        if (!backupMetadataOptional.isPresent()) {
             object.put("Snapshotstatus", false);
-            String token =
-                    SystemUtils.getDataFromUrl(
-                            "http://localhost:8080/Priam/REST/v1/cassconfig/get_token");
-            if (token != null && !token.isEmpty()) {
-                object.put("token", token);
-            } else {
-                object.put("token", "not available");
-            }
-        }
+        } else {
 
+            object.put("Snapshotstatus", true);
```

Comment:
Did we not rely on the `token`, `start_time`, `complete_time` etc? 

If we are breaking backwards compatibility using snake case in json is really nice (e.g. `snapshot_status`, `details` etc ...)

Reply:
no we do not rely on other fields but "Snapshotstatus". Thus, not breaking the compatibility by keeping that field only. All other fields were added without any documentation as well so no one should rely on that.